src/YarpUtilities/include/BipedalLocomotionControllers/YarpUtilities/Helper.tpp

```diff
@@ -0,0 +1,238 @@
+                      << key << " is not a " << YARP_UTILITES_GET_ELEMENT_TYPE(elementType) << "."
+                      << std::endl;
+            return false;
+        }
+
+        vector[i] = convertValue<elementType>(inputPtr->get(i));
+    }
+    return true;
+}
+
+template <>
+bool getVectorFromSearchable<std::vector<bool>>(const yarp::os::Searchable& config,
+                                                const std::string& key,
+                                                std::vector<bool>& vector)
+{
+
+    static_assert(YARP_UTILITES_CHECK_ELEMENT_SUPPORT(bool),
+                  "[BipedalLocomotionControllers::YarpUtilities::getVectorFromSearchable] The "
+                  "function getElementFromSearchable() cannot be called with the desired "
+                  "element type");
+
+    yarp::os::Value* value;
+    if (!config.check(key, value))
+    {
+        std::cerr << "[BipedalLocomotionControllers::YarpUtilities::getVectorFromSearchable] "
+                     "Missing field "
+                  << key << std::endl;
+        return false;
+    }
+
+    if (value->isNull())
+    {
+        std::cerr << "[BipedalLocomotionControllers::YarpUtilities::getVectorFromSearchable] Empty "
+                     "input value named "
+                  << key << std::endl;
+        return false;
+    }
+
+    if (!value->isList())
+    {
+        std::cerr << "[BipedalLocomotionControllers::YarpUtilities::getVectorFromSearchable] The "
+                     "value named "
+                  << key << "is not associated to a list." << std::endl;
+        return false;
+    }
+
+    yarp::os::Bottle* inputPtr = value->asList();
+    if (inputPtr == nullptr)
+    {
+        std::cerr << "[BipedalLocomotionControllers::YarpUtilities::getVectorFromSearchable] The "
+                     "list associated to the value named "
+                  << key << " is empty." << std::endl;
+        return false;
+    }
+
+    // resize the vector
+    vector.resize(inputPtr->size());
+
+    for (int i = 0; i < inputPtr->size(); i++)
+    {
+        if (!(inputPtr->get(i).isBool()) && !(inputPtr->get(i).isInt()))
+        {
+            std::cerr << "[BipedalLocomotionControllers::YarpUtilities::getVectorFromSearchable] "
+                         "The element of the list associated to the value named "
+                      << key << " is not a boolean ." << std::endl;
+            return false;
+        }
+
+        vector[i] = convertValue<bool>(inputPtr->get(i));
+    }
+    return true;
+}
+
+template <typename T> void mergeSigVector(yarp::sig::Vector& vector, const T& t)
+{
+    for (int i = 0; i < t.size(); i++)
+        vector.push_back(t(i));
+
+    return;
+}
+
+template <typename T, typename... Args>
+void mergeSigVector(yarp::sig::Vector& vector, const T& t, const Args&... args)
+{
+    for (int i = 0; i < t.size(); i++)
+        vector.push_back(t(i));
```

Comment:
Why not ``mergeSigVector(vector, t)``?

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-controllers/pull/5/commits/1b16718e70cc3fb6687c0cd6249ccdb6d97a2f49